packages/client/wallets/aa/src/services/logging/DatadogProvider.ts

```diff
@@ -0,0 +1,39 @@
+import { datadogLogs } from "@datadog/browser-logs";
+
+
+import { BrowserLoggerInterface } from "./BrowserLoggerInterface";
+import { DATADOG_CLIENT_TOKEN } from "@/utils/constants";
+
+type DatadogContext = object | undefined;
+
+export class DatadogProvider implements BrowserLoggerInterface {
+    logInfo(message: string, context?: unknown) {
+        const _context = context as DatadogContext;
+        init();
+        datadogLogs.logger.info(message, _context);
+    }
+    logError(message: string, context?: unknown) {
+        const _context = context as DatadogContext;
+        init();
+        datadogLogs.logger.error(message, _context);
+    }
+    logWarn(message: string, context?: unknown) {
+        const _context = context as DatadogContext;
+        init();
+        datadogLogs.logger.warn(message, _context);
+    }
+}
```

Comment:
we've got a lot of repetition between these functions, can we instead use the constructor?
```
export class DatadogProvider implements BrowserLoggerInterface {
  constructor() {
    super();
    init();
  }
  
  logError(message: string, context?: unknown) {
        datadogLogs.logger.error(message, context as DatadogContext);
  }
  
  ...